src/main/java/org/terasology/las/bases/BaseProvider.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.las.bases;
+
+import com.google.common.collect.ImmutableSet;
+import org.terasology.math.Region3i;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.world.generation.*;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+import java.util.Collection;
+
+@Produces(BaseFacet.class)
+@Requires(@Facet(SurfaceHeightFacet.class))
+
+public class BaseProvider implements FacetProvider {
+    int baseExtent = 2; //determines size of base (# of tiles from center)
```

Comment:
I guess this is all WIP, right? Otherwise, define the `BASE_EXTEND` as constant (similarly for the base center positions)

Reply:
What do I need to do to turn things into javadoc? I saw this (https://github.com/MovingBlocks/Terasology/wiki/JavaDoc) but wasn't sure about it.

Reply:
Easy! Literally write it `/** Like this */` instead of `// Like this`

Usually I like the one liner for field variables that can be described really easily. For longer stuff or methods use the multiline version:

```
/**
 * Stuff.
 */
```